cpp/src/arrow/type.h

```diff
@@ -890,6 +950,15 @@ std::shared_ptr<DataType> list(const std::shared_ptr<Field>& value_type);
 ARROW_EXPORT
 std::shared_ptr<DataType> list(const std::shared_ptr<DataType>& value_type);
 
+/// \brief Return an Duration instance.
+std::shared_ptr<DataType> ARROW_EXPORT duration_interval(TimeUnit::type unit);
+
+/// \brief Return an DayTimeIntervalType instance
+std::shared_ptr<DataType> ARROW_EXPORT day_time_interval();
+
+/// \brief Return an MonthIntervalType instance
+std::shared_ptr<DataType> ARROW_EXPORT month_interval();
```

Comment:
nit: as far as naming I could see an argument for using

* `interval_duration`
* `interval_day_time`
* `interval_month`

but I'm OK with the way things are here

Reply:
I could see this to, but I'm happier without the reverse notation.

Reply:
note "duration" conflicts with some of our time classes so I renamed to "duration_type", hopefully that is ok

Reply:
"some of our time classes"?

Reply:
Need to look back I think it was in the vendored time library we were using.  I can look deeper if you don't think there should be a conflict

Reply:
looking again they shouldn't conflict, I will try to fix this today.

Reply:
Awesome, thanks.

Reply:
renamed.  had to touch some of the vendored code.